### PR TITLE
Remove validator vote signatures from block

### PR DESCRIPTION
**Pull request by jinmannwong**
- Refactor types.Commit into two separate structs types.VotesCommit and types.BlockCommit. The first is used for saving to store the vote messages seen for a block and contains all the signatures in a vote message. The latter is the commit included in a block that does not contain any signature from individual validators, but instead contains the combined signature
- Also removed types.CommitSig and separated this into types.CommitSigVote and types.CommitSigBlock, where the first contains the validator signatures and the latter does not.

### CODE REVIEW

File: consensus/common_test.go [hunk 1]
```diff
@@ -503,7 +503,7 @@ func ensureNewRound(roundCh <-chan tmpubsub.Message, height int64, round int) {
 }
 
 func ensureNewTimeout(timeoutCh <-chan tmpubsub.Message, height int64, round int, timeout int64) {
-	timeoutDuration := time.Duration(timeout*10) * time.Nanosecond
+	timeoutDuration := time.Duration(timeout*100) * time.Nanosecond
```
Comment (jinmannwong) [inline consensus/common_test.go:506]: Increased this timeout to fix a flakey test

File: consensus/state_test.go [hunk 2]
```diff
@@ -1528,12 +1528,11 @@ func TestStartNextHeightCorrectly(t *testing.T) {
 	// add precommits
 	signAddVotes(cs1, types.PrecommitType, nil, types.PartSetHeader{}, vs2)
 	signAddVotes(cs1, types.PrecommitType, theBlockHash, theBlockParts, vs3)
-	time.Sleep(5 * time.Millisecond)
+	assert.Eventually(t, func() bool {
+		rs = cs1.GetRoundState()
+		return rs.TriggeredTimeoutPrecommit
+	}, 20*time.Millisecond, 1*time.Millisecond)
```
Comment (jinmannwong) [inline consensus/state_test.go:1534]: Again, this is to hopefully fix a flakey test

File: types/prevote_set.go [hunk 3]
```diff
@@ -593,16 +586,3 @@ func (vs *blockVotes) getByIndex(index int) *Vote {
...
-// Common interface between *consensus.VoteSet and types.Commit
-type VoteSetReader interface {
-	GetHeight() int64
-	GetRound() int
-	Type() byte
-	Size() int
-	BitArray() *bits.BitArray
-	GetByIndex(int) *Vote
-	IsCommit() bool
-}
```
Comment (jinmannwong) [inline types/prevote_set.go:608]: Note required anymore

File: state/services.go [hunk 4]
```diff
@@ -21,15 +21,15 @@ type BlockStore interface {
...
 
 	PruneBlocks(height int64) (uint64, error)
 
 	LoadBlockByHash(hash []byte) *types.Block
 	LoadBlockPart(height int64, index int) *types.Part
 
-	LoadBlockCommit(height int64) *types.Commit
-	LoadSeenCommit(height int64) *types.Commit
+	LoadBlockCommit(height int64) *types.BlockCommit
+	LoadSeenCommit(height int64) types.SeenCommit
```
Comment (n-hutton) [inline state/services.go:32]: why is it switching from pointer to non pointer?

**Review by n-hutton — APPROVED** [review]

File: state/services.go (hunk 4 above)
Comment (jinmannwong) [reply state/services.go:32]: Ah, because `types.SeenCommit` is an interface.

**Pull request merged**